lib/site.plugin.js

```diff
@@ -1,38 +1,134 @@
 /**
- * Module variables
+ * Module vars
  */
 const root = '/sites';
 
 class SitePlugin {
+
 	/**
 	 * `SitePlugin` constructor.
 	 *
-	 * @param {String} [id] - the plugin ID
+	 * @param {String} [slug] - the plugin slug
 	 * @param {Number|String} sid - site identifier
 	 * @param {WPCOM} wpcom - wpcom instance
 	 * @return {Undefined} undefined
 	 */
-	constructor( id, sid, wpcom ) {
+	constructor( slug, sid, wpcom ) {
 		if ( ! ( this instanceof SitePlugin ) ) {
-			return new SitePlugin( id, sid, wpcom );
+			return new SitePlugin( slug, sid, wpcom );
 		}
 
-		this._id = id;
+		this._slug = encodeURIComponent( slug );
 		this._sid = sid;
-		this.path = `${root}/${this._sid}/plugins`;
 		this.wpcom = wpcom;
+
+		this.path = `${root}/${ this._sid }/plugins`;
+		this.pluginPath = `${ this.path }/${ this._slug }`;
```

Comment:
Seems like since only `this.pluginPath` is used below, these two lines could be combined to create `this.pluginPath` here.